src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -0,0 +1,421 @@
+                    //Printing packages of a single project and keeping track if
+                    //an auto-referenced package was printed
+                    var autoRefFoundWithinProject = false;
+                    PrintProjectPackages(packages, projectName, listPackageArgs.IncludeTransitive, listPackageArgs.IncludeOutdated, out autoRefFoundWithinProject);
+                    autoReferenceFound = autoReferenceFound || autoRefFoundWithinProject;
+                }
+
+                //Unload project
+                ProjectCollection.GlobalProjectCollection.UnloadProject(project);
+            }
+
+            //If any auto-references were found, a line is printed
+            //explaining what (A) means
+            if (autoReferenceFound)
+            {
+                Console.WriteLine(Strings.ListPkg_AutoReferenceDescription);
+            }
+        }
+
+        /// <summary>
+        /// Fetches the latest versions for all of the packages that are
+        /// to be listed
+        /// </summary>
+        /// <param name="packages">The packages found in a project</param>
+        /// <param name="listPackageArgs">List args for the token and source provider</param>
+        /// <returns>A data structure like packages, but includes the latest versions</returns>
+        private async Task AddLatestVersionsAsync(
+            IEnumerable<FrameworkPackages> packages, ListPackageArgs listPackageArgs)
+        {
+            var providers = Repository.Provider.GetCoreV3();
+
+            foreach (var frameworkPackages in packages)
+            {
+                var topLevelTasks = new List<Task>();
+                foreach (var topLevelPackage in frameworkPackages.TopLevelPackages)
+                {
+                    topLevelTasks.Add(GetLatestVersionAsync(topLevelPackage, NuGetFramework.Parse(frameworkPackages.Framework), listPackageArgs, providers));
+                }
+
+                var transitiveTasks = new List<Task>();
+                foreach (var transitivePackage in frameworkPackages.TransitivePacakges)
+                {
+                    topLevelTasks.Add(GetLatestVersionAsync(transitivePackage, NuGetFramework.Parse(frameworkPackages.Framework), listPackageArgs, providers));
+                }
+
+                await Task.WhenAll(topLevelTasks);
+                await Task.WhenAll(transitiveTasks);
+            }
+        }
+
+        /// <summary>
+        /// Fetches the latest version of the given package
+        /// </summary>
+        /// <param name="package">The package to get the latest version for</param>
+        /// <param name="framework">The framework for the given package</param>
+        /// <param name="listPackageArgs">List args for the token and source provider></param>
+        /// <param name="providers">The providers to use when looking at sources</param>
+        /// <returns>An updated pacakge with the highest version at sources</returns>
+        private async Task GetLatestVersionAsync(
```

Comment:
naming: It doesn't make sense to call the method `GetLatestVersionAsync` if it doesn't return a version.